SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -66,43 +66,55 @@ void USpatialSender::Init(USpatialNetDriver* InNetDriver)
 	ClassInfoManager = InNetDriver->ClassInfoManager;
 }
 
+const WorkerAttributeSet ServerWorkerAttributeSet = { SpatialConstants::ServerWorkerType };
+const WorkerAttributeSet ClientWorkerAttributeSet = { SpatialConstants::ClientWorkerType };
+
+const WorkerRequirementSet ServerWorkerOnlyRequirementSet = { ServerWorkerAttributeSet };
+
 Worker_RequestId USpatialSender::CreateEntity(USpatialActorChannel* Channel)
 {
 	AActor* Actor = Channel->Actor;
 	UClass* Class = Actor->GetClass();
 
 	FString ClientWorkerAttribute = improbable::GetOwnerWorkerAttribute(Actor);
 
-	WorkerAttributeSet ServerAttribute = { SpatialConstants::ServerWorkerType };
-	WorkerAttributeSet AIWorkerAttribute = { FString(TEXT("AIWorker")) };
-	WorkerAttributeSet ClientAttribute = { SpatialConstants::ClientWorkerType };
-	WorkerAttributeSet OwningClientAttribute = { ClientWorkerAttribute };
+	TArray<FString> ServerWorkerTypes = GetDefault<USpatialGDKSettings>()->ServerWorkerTypes;
 
-	WorkerRequirementSet ServersOnly = { ServerAttribute };
-	WorkerRequirementSet AllServerTypes = { ServerAttribute, AIWorkerAttribute };
-	WorkerRequirementSet ClientsOnly = { ClientAttribute };
-	WorkerRequirementSet OwningClientOnly = { OwningClientAttribute };
+	WorkerRequirementSet AllServerTypesRequirementSet;
+	WorkerRequirementSet AnyServerOrClientRequirementSet = { ClientWorkerAttributeSet };
+
+	WorkerAttributeSet OwningClientAttributeSet = { ClientWorkerAttribute };
+
+	WorkerRequirementSet AnyServerOrOwningClientRequirementSet = { OwningClientAttributeSet };
+	WorkerRequirementSet OwningClientOnlyRequirementSet = { OwningClientAttributeSet };
+
+	for(auto i=0; i < ServerWorkerTypes.Num(); ++i)
```

Comment:
```suggestion
	for(uint32 i=0; i < ServerWorkerTypes.Num(); ++i)
```

Reply:
Could this be a `for(element : container)` loop? do we need the index?

Reply:
ah yes, doing the latter